src/browser/xhr/form_data.zig

```diff
@@ -186,9 +199,167 @@ const EntryIterator = struct {
+                                .value = "0",
+                            });
+                        }
+                    }
+                    continue;
+                }
+                if (std.ascii.eqlIgnoreCase(input_type, "checkbox") or std.ascii.eqlIgnoreCase(input_type, "radio")) {
+                    if (try parser.inputGetChecked(@ptrCast(element)) == false) {
+                        continue;
+                    }
+                }
+                const value = (try parser.elementGetAttribute(element, "value")) orelse "";
+                try entries.append(arena, .{ .key = name, .value = value });
+            },
+            .select => {
+                const select: *parser.Select = @ptrCast(node);
+                try collectSelectValues(arena, select, name, &entries, state);
+            },
+            .textarea => {
+                const textarea: *parser.TextArea = @ptrCast(node);
+                const value = try parser.textareaGetValue(textarea);
+                try entries.append(arena, .{ .key = name, .value = value });
+            },
+            .button => if (submitter_name_) |submitter_name| {
+                if (std.mem.eql(u8, submitter_name, name)) {
+                    const value = (try parser.elementGetAttribute(element, "value")) orelse "";
+                    try entries.append(arena, .{ .key = name, .value = value });
+                }
+            },
+            else => {
+                log.warn("unsupported form element: {s}\n", .{@tagName(tag)});
+                continue;
+            },
+        }
+    }
+
+    return entries;
+}
+
+fn collectSelectValues(arena: Allocator, select: *parser.Select, name: []const u8, entries: *std.ArrayListUnmanaged(Entry), state: *SessionState) !void {
+    const HTMLSelectElement = @import("../html/select.zig").HTMLSelectElement;
+
+    // Go through the HTMLSelectElement because it has specific logic for handling
+    // the default selected option, which libdom doesn't properly handle
+    const selected_index = try HTMLSelectElement.get_selectedIndex(select, state);
+    if (selected_index == -1) {
+        return;
+    }
+    std.debug.assert(selected_index >= 0);
+
+    const options = try parser.selectGetOptions(select);
+    const is_multiple = try parser.selectGetMultiple(select);
+    if (is_multiple == false) {
+        const option = try parser.optionCollectionItem(options, @intCast(selected_index));
+        const value = try parser.optionGetValue(option);
+        return entries.append(arena, .{ .key = name, .value = value });
+    }
+
+    const len = try parser.optionCollectionGetLength(options);
+
+    // we can go directly to the first one
+    for (@intCast(selected_index)..len) |i| {
+        const option = try parser.optionCollectionItem(options, @intCast(i));
+        if (try parser.elementGetAttribute(@ptrCast(option), "disabled") != null) {
```

Comment:
You don't check disable attr in the single select case. Is it intentional?

Reply:
Nope, I missed it. I checked the browser, if the default item (the first one) is disabled, then it sends nothing. Fixed in https://github.com/lightpanda-io/browser/pull/669/commits/72944a4e5e1cac4ba6ad6b3ffa34a4ed781bc402